tests/core/test_dataset_filters.py

```diff
@@ -2326,27 +2326,79 @@ def test_extract_values_open_intervals(grid4x4):
 
 @pytest.fixture()
 def labeled_data():
-    sphere = pv.Sphere(radius=0.25)
-    box = pv.Box(bounds=(-1.0, 1.0, -1.0, 1.0, -1.0, 1.0))
-    labeled = (sphere + box).extract_geometry().connectivity()
+    bounds = np.array((-0.5, 0.5, -0.5, 0.5, -0.5, 0.5))
+    small_box = pv.Box(bounds=bounds)
+    big_box = pv.Box(bounds=bounds * 2)
+    labeled = (small_box + big_box).extract_geometry().connectivity()
     assert isinstance(labeled, pv.PolyData)
     assert labeled.array_names == ['RegionId', 'RegionId']
-    assert box.volume > 0
-    assert sphere.volume > 0
-    return sphere, box, labeled
+    assert np.allclose(small_box.volume, 1**3)
+    assert np.allclose(big_box.volume, 2**3)
+    return small_box, big_box, labeled
 
 
-def test_extract_values_split(labeled_data):
-    sphere, box, labeled = labeled_data
-    multiblock = labeled.extract_values()
-    assert len(multiblock) == 2
-    assert isinstance(multiblock[0], pv.UnstructuredGrid)
-    assert isinstance(multiblock[1], pv.UnstructuredGrid)
+def add_component_to_labeled_data(labeled_data, offset):
+    """Add second component to scalars by duplicating first component and adding offset."""
+    small_box, big_box, labeled_data = labeled_data
+
+    point_data = labeled_data.point_data['RegionId']
+    point_data = np.vstack([point_data, point_data + offset]).T
+    labeled_data.point_data['RegionId'] = point_data
+
+    cell_data = labeled_data.cell_data['RegionId']
+    cell_data = np.vstack([cell_data, cell_data + offset]).T
+    labeled_data.cell_data['RegionId'] = cell_data
+
+    return small_box, big_box, labeled_data
+
+
+@pytest.mark.parametrize('component_offset', [0, -0.5, None])
+@pytest.mark.parametrize('component', [0, 1, 'any', 'all'])
+def test_extract_values_component_split(labeled_data, component_offset, component):
```

Comment:
https://github.com/pyvista/pyvista/pull/6001#pullrequestreview-2033197591 is saying this function.

Reply:
This test has been updated to remove the if-else blocks. if-else has been replaced with explicit test case tuples which map the inputs to the expected outputs.